### DISCUSSION TITLE
Command not found

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#697] [Q&A] [answered]
I've got a CLI I installed via homebrew to resize my displays properly: https://github.com/jakehilborn/displayplacer

When I try to use this with ScriptKit I get an error that it can't find the CLI. I notice when using `exec` it tries to use `/bin/sh` instead of `/bin/zsh` and I'm guessing my `.zshrc` config isn't being used either so my environment variables are not set up like they are in my terminal.

What's the proper way to solve this?

```
[2022-01-25 21:25:54.294] [warn]  ☠️ ERROR PROMPT SHOULD SHOW ☠️
[2022-01-25 21:25:54.295] [warn]  Error: Command failed with exit code 127: displayplacer "id:14EB5859-EA7B-4C0A-BE57-3BEFA8CB0C09 res:2560x1440 hz:60 color_depth:8 scaling:on origin:(0,0) degree:0" "id:3867FCD7-633C-4657-81A4-EBF454F8DD84 res:1920x1080 hz:60 color_depth:8 scaling:on origin:(2560,0) degree:0"
/bin/sh: displayplacer: command not found
    at makeError (file:///Users/kentcdodds/.kit/node_modules/@johnlindquist/globals/dist/index.js:15:247182)
    at file:///Users/kentcdodds/.kit/node_modules/@johnlindquist/globals/dist/index.js:15:258269
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async file:///Users/kentcdodds/.kenv/.scripts/fix-screens.mjs:3:14
```

```ts
// Menu: Fix Screens
// Description: Set screens to the right size
// Author: Kent C. Dodds
// Shortcut: command option control f
// Twitter: @kentcdodds

import "@johnlindquist/kit";

await exec(
  'displayplacer "id:14EB5859-EA7B-4C0A-BE57-3BEFA8CB0C09 res:2560x1440 hz:60 color_depth:8 scaling:on origin:(0,0) degree:0" "id:3867FCD7-633C-4657-81A4-EBF454F8DD84 res:1920x1080 hz:60 color_depth:8 scaling:on origin:(2560,0) degree:0"'
);
```

**Comment by johnlindquist** [comment, marked as answer]
## Option 1 - Use full tool path

1. In your terminal, run `which displayplacer | pbcopy`. 
2. This will copy `/opt/homebrew/bin/displayplacer` to the clipboard.
3. Use full path
```js
await exec(`/opt/homebrew/bin/displayplacer "blah blah blah"`)
```

### Probably faster 🚗

Using the full path skips the "lookup" process on the PATH for the tool. Some tools, like nvm, really slow down looking up tools on the PATH. For example, if you ever find running `code` in your terminal is launching slowly, it's probably nvm's fault for messing with the PATH.

## Option 2 - Sync profile path

In your terminal run the following command:

```bash
~/.kit/bin/kit sync-path
```

This will help sync the PATH your terminal uses with the PATH that Script Kit should use.

<img width="822" alt="CleanShot 2022-01-25 at 21 52 35@2x" src="https://user-images.githubusercontent.com/36073/151105952-2f58689c-32aa-4323-a70b-f23d0b55d0b5.png">

## Why?

Each time you open a terminal, it loads a bash/zsh/shell profile that modifies the PATH. Script Kit doesn't know the user's preferred shell/profile, so the PATH is set to the defaults. Looking up tools installed by brew, etc will be missing since brew doesn't use a default PATH. I'm definitely considering adding brew to the Script Kit PATH since this question has popped up a few times.

**Reply by kentcdodds** [reply]
First option did it just fine. Sorry, I did try to search for an existing answer but couldn't find anything 😅 Thanks John!

This is going to be *soooo* nice because my displays are always messed up when I start my machine up after sleeping or plug it into the displays after a while.

**Reply by johnlindquist** [reply]
I was playing around with a Display Placer script last year. This might be helpful:

```js
//Menu: Change Resolution

let {stdout} = await exec(`/opt/homebrew/bin/displayplacer list | grep displayplacer.*res`)

let displays = stdout.trim()
  .slice("displayplacer".length)
  .replaceAll('"', "")
  .trim()
  .replace(" id", "||id")
  .split("||")
  .map(string =>
    string.split(" ").reduce((acc, string) => {
      let [key, value] = string.split(":")
      return { ...acc, [key]: value }
    }, {})
  )

let displayId = await arg("Select display:", {
  choices: displays.map(
    ({ id, res, hz, scaling, origin }) => {
      return {
        name: `${res} - scaling:${scaling} - origin:${origin}`,
        value: id,
      }
    }
  ),
})

let changeResolution = (
  id,
  width,
  height,
  scaling = "off"
) => {
  exec(
    `/opt/homebrew/bin/displayplacer "id:${id} res:${width}x${height}" scaling:${scaling}`
  )
}

let resolution = await arg("Resolution:", {
  choices: [
    {
      name: "1280x720",
      value: { width: 1280, height: 720, scaling: "off" },
    },
    {
      name: "1920x1080",
      value: { width: 1920, height: 1080, scaling: "off" },
    },
    {
      name: "1280x720 HiDPI",
      value: { width: 1280, height: 720, scaling: "on" },
    },
    {
      name: "1920x1080 HiDPI",
      value: { width: 1920, height: 1080, scaling: "off" },
    },
  ],
})

changeResolution(displayId, 1280, 720, "on")
```
